Program.cs

```diff
@@ -9,6 +9,7 @@ static void Main(string[] args)
             Console.WriteLine("Hello World!");
```

Comment:
* if the input is null, it throws an exception
* if the sub text is null, it throws an exception
* line 31: for (int i = 0; i < text.Length && i < (text.Length - subText.Length +1); i++)
the condition is not clear, add a variable with the condition with appropriate name.
* rename the variables i and q to an appropriate names.
* What is the Big O notation for time for the current code.
* Can you rewrite the code to reduce the time complexity of the code (Big O notation)? if yes, please apply your changes.

Reply:
* Handled nulls 
* Renamed loop variables
* small enhancement for outer Loop to be limited by subtext length and available room
*  Complexity O(n*m) where n is length of text, m is length of subtext, don't think I can improve that without dropping an  edge case, I can rewrite to be better if "text" is filled with random characters, but it would fail if all characters are the same one.

Reply:
I believe you can reduce the complexity to O(n) by removing the inner loop.
think about it, and let me know if you figured a way to do it.